### DISCUSSION TITLE
Read State with Tasmota Script and combine with &

### DISCUSSION THREAD

**Discussion by Landixus** [arendst/Tasmota#22485] [Q&A]
I update a bit:
I try to switch 2 PowerPlugs ON OFF with this script
`>D 48
pcurr=0
wcurr=0
;IP der S20 Steckdose
url2="[192.168.0.92]"
url1="[192.168.0.161]"
state1=""
state2=""
hgp=0
gp=0.299

>T
pcurr=Haus#current
>S
if upsecs%10==0
then
WebQuery %url1%/cm?cmnd=Power GET
WebQuery %url2%/cm?cmnd=Power GET

if pcurr<-180 & state1"OFF"
then
=>WebSend %url1% POWER1 ON
endif
if pcurr>100 & state1"ON"
then
=>WebSend %url1% POWER1 OFF
endif
if pcurr<-200 & state2"ON"
then
=>WebSend %url2% POWER1 ON
endif
if pcurr>100
then
=>WebSend %url2% POWER1 OFF
endif
hgp=(wcurr*gp)
>E
state1=WebQuery#POWER
state2=WebQuery#POWER
print state1 %state1%
print state2 %state2%
>B
=>sensor53 r
>M 2
+1,3,o,0,9600,Haus
+2,1,o,0,9600,Haus

1,1-0:1.8.0*255(@1,Gesamtverbrauch,KWh,total,4
1,1-0:1.8.0*96(@1,Verbrauch 1 Tag,KWh,total_1d,4
1,1-0:1.8.0*97(@1,Verbrauch 7 Tage,KWh,total_7d,4
1,1-0:1.8.0*98(@1,Verbrauch 30 Tage,KWh,total_30d,4
1,1-0:1.8.0*99(@1,Verbrauch 1 Jahr,KWh,total_365d,4
1,1-0:16.7.0*255(@1,Verbrauch aktuell,W,current,17
1,1-0:2.8.0*255(@1,Gesamt Verschenkt,KWh,totalV,4
print %Pa%
>W
===========
Verschenkt in Euro {m} %2hgp% €
#`

State is NOW:
`22:06:40.589 state1   ? ` ?( ?ȱ&@
22:06:40.591 state2   ? ` ?( ?ȱ&@`
But its not readable this way
Console on WebRequest
with 
`WebQuery http://192.168.0.92/cm?cmnd=Power GET`
gives
`16:04:12.224 MQT: stat/stromzaehler/RESULT = {"WebQuery":{"POWER":"OFF"}}
16:04:12.232 MQT: stat/stromzaehler/RESULT = {"WebSend":"Done"}`

Help pls

**Comment by sfromis** [comment]
I can't answer about the scripter, but I noticed a detail (twice) looking odd:
`if pcurr<-180 & state1"OFF"`
I'd expect a comparison operator to be needed there.

**Comment by Landixus** [comment]
I am not sure, i get an output, but is not readable, and state1"OFF" should be the memory VAR, so there is no operator needed?
mayBE i need to convert somewhere, but i am a bit lost here.

**Reply by sfromis** [reply]
What I meant was that you have a comparison operator with `pcurr`, but none for `state1`, where I'd expect something like when the docs examples are like `if istr=="0x00F7C03F"` - correct syntax for a comparison. Don't expect it to work smoothly if your code has syntax errors.

**Reply by Landixus** [reply]
uff, i dont understand!?

**Reply by sfromis** [reply]
You have no comparison operator, like `==` in the quoted example from the docs. Just two values adjacent is not a comparison. I'd expect the valid to be `state1=="OFF"` instead of `state1"OFF"`

The docs lists: Comparison operators `==`, `!=`, `>`, `>=`, `<`, `<=`

**Reply by Landixus** [reply]
ah ok, now i understand, but i was so upset of my faults and installed on my NUC with iOBroker Javascript with rules.
Now it works, but i want not to jailed by the NUC, and  go deeper in it when i find more time

**Comment by Landixus** [comment]
I make progress:
`>T
pcurr=Haus#current
>S
if upsecs%10==0
then
WebQuery %url1%/cm?cmnd=Power GET
print CALLED --->
;<-180
if pcurr<-180 & %state1%"0.00"
then
=>WebSend %url1% POWER1 ON
endif
if pcurr>100 & %state1%"1.00"
then
=>WebSend %url1% POWER1 OFF
endif
hgp=(wcurr*gp)
print state1 %state1%
>E
state1=WebQuery#POWER`

I can read the state and switch, now i need to bring the second plug inGame.
But i am still not sure that it is correct!?

**Comment by Landixus** [comment]
When i add the second plug, both states are 0.00 , also when plug1 is ON :(
`>S
if upsecs%10==0
then
WebQuery %url1%/cm?cmnd=Power GET
state1=WebQuery#POWER 
WebQuery %url2%/cm?cmnd=Power GET
state2=WebQuery#POWER 
print CALLED --->
;<-180
if pcurr<-180 & %state1%"0.00"
then
=>WebSend %url1% POWER1 ON
endif
if pcurr>100 & %state1%"1.00"
then
=>WebSend %url1% POWER1 OFF
endif
;new Plug
if pcurr<-200 & %state1%"1.00"
then
=>WebSend %url2% POWER1 ON
endif
if pcurr>100 & %state1%"1.00"
then
=>WebSend %url2% POWER1 OFF
endif


hgp=(wcurr*gp)
print "state1: " %state1%
print "state2: " %state2%
>E
state1=WebQuery#POWER
state2=WebQuery#POWER
>B...`
